src/core/users/sagas/searchAllUsers.js

```diff
@@ -0,0 +1,73 @@
+/*
+ * @flow
+ */
+
+import isEmail from 'validator/lib/isEmail';
+import { call, put, takeEvery } from '@redux-saga/core/effects';
+import { PrincipalsApiActions, PrincipalsApiSagas } from 'lattice-sagas';
+import { AxiosUtils, LangUtils, Logger } from 'lattice-utils';
+import type { Saga } from '@redux-saga/core';
+import type { WorkerResponse } from 'lattice-sagas';
+import type { SequenceAction } from 'redux-reqseq';
+
+import { SEARCH_ALL_USERS, searchAllUsers } from '../actions';
+
+const { searchUsers } = PrincipalsApiActions;
+const { searchUsersWorker } = PrincipalsApiSagas;
+const { toSagaError } = AxiosUtils;
+const { isNonEmptyString } = LangUtils;
+
+const LOG = new Logger('UsersSagas');
+
+function* searchAllUsersWorker(action :SequenceAction) :Saga<void> {
+
+  try {
+    yield put(searchAllUsers.request(action.id, action.value));
+
+    if (action.value === '*') {
+      throw new Error('search query cannot be "*"');
+    }
+    if (!isNonEmptyString(action.value)) {
+      throw new Error('search query cannot be empty');
+    }
```

Comment:
no change required, but I think it feels more natural for these two statements to be reversed (First verify that the value is a non-empty string, then verify the string is not of value *)

Reply:
Fixed.